feeds/scheduler/scheduler.go

```diff
@@ -27,39 +35,35 @@ type pollResult struct {
 	err      error
 }
 
-// Poll fetches the latest packages from each registered feed.
-func (s *Scheduler) Poll(cutoff time.Time) ([]*feeds.Package, []error) {
-	results := make(chan pollResult)
-	for name, feed := range s.registry {
-		go func(name string, feed feeds.ScheduledFeed) {
-			result := pollResult{
-				name: name,
-				feed: feed,
-			}
-			result.packages, result.err = feed.Latest(cutoff)
-			results <- result
-		}(name, feed)
+// Runs several services for the operation of scheduler, this call is blocking until application exit
+// or failure in the HTTP server
+// Services include: Cron polling via FeedGroups, HTTP serving of PollHandler.
+func (s *Scheduler) Run(initialCutoff time.Duration, enableDefaultTimer bool) error {
+	defaultSchedule := fmt.Sprintf("@every %s", initialCutoff.String())
+
+	allFeeds := []feeds.ScheduledFeed{}
+	for _, feed := range s.registry {
+		allFeeds = append(allFeeds, feed)
 	}
-	errs := []error{}
-	packages := []*feeds.Package{}
-	for i := 0; i < len(s.registry); i++ {
-		result := <-results
 
-		logger := log.WithField("feed", result.name)
-		if result.err != nil {
-			logger.WithError(result.err).Error("error fetching packages")
-			errs = append(errs, result.err)
-			continue
-		}
-		for _, pkg := range result.packages {
-			log.WithFields(log.Fields{
-				"feed":    result.name,
-				"name":    pkg.Name,
-				"version": pkg.Version,
-			}).Print("Processing Package")
+	// Configure cron job for scheduled polling
+	cronJob := cron.New()
+	feedGroup := NewFeedGroup(allFeeds, s.publisher, initialCutoff)
+	if enableDefaultTimer {
+		err := cronJob.AddJob(defaultSchedule, feedGroup)
```

Comment:
Main currently logs/prints that it's running a timer at a given schedule, can we add this here too?